cmake/opendds_build_helpers.cmake

```diff
@@ -18,6 +18,21 @@ function(_opendds_alias target)
     add_executable("${name}" ALIAS "${target}")
   endif()
   set_target_properties(${target} PROPERTIES EXPORT_NAME "${name}")
+
+  if(OPENDDS_COMPILE_WARNINGS)
+    if(CMAKE_CXX_COMPILER_ID MATCHES "Clang" OR CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+      target_compile_options(${target} PRIVATE
+        -Wall -Wpedantic -Wno-unused -Wextra -Wcast-align -Wcast-qual -Wctor-dtor-privacy
+        -Wdisabled-optimization -Wformat=2 -Winit-self -Wmissing-include-dirs -Woverloaded-virtual
+        -Wredundant-decls -Wshadow -Wsign-conversion -Wstrict-overflow=5 -Wundef
+      )
+      if(OPENDDS_COMPILE_WARNINGS STREQUAL "ERROR")
+        target_compile_options(${target} PRIVATE -Werror)
+      endif()
```

Comment:
This is missing on the MSVC branch
See https://learn.microsoft.com/en-us/cpp/build/reference/compiler-option-warning-level?view=msvc-170

Reply:
Are you talking about `/WX`?